lib/auth/googleauth.js

```diff
@@ -294,12 +309,25 @@ GoogleAuth.prototype.fromJSON = function(json, opt_callback) {
  */
 GoogleAuth.prototype.fromStream = function(stream, opt_callback) {
   var that = this;
-  var jwtClient = new that.JWTClient();
-  jwtClient.fromStream(stream, function(err) {
-    if (err) {
+  if (!stream) {
+    process.nextTick(function() {
+      callback(
+        opt_callback,
+        new Error('Must pass in a stream containing the service account auth settings.'));
+    });
+    return;
+  }
+  var s = '';
```

Comment:
I believe this stream handling code exists in three places now: googleauth, jwtclient, and refreshclient. Probably this code should be pushed down into authclient (or oauthclient) and jwtclient and refreshclient should call down to the base class. Also googleauth should delegate the fromStream call down to either jwtclient or refreshclient, similar to what happens in fromJSON.